hivemq-edge/src/main/java/com/hivemq/protocols/ProtocolAdapterManager.java

```diff
@@ -150,21 +176,32 @@ public void refresh(final @NotNull List<ProtocolAdapterEntity> configs) {
 
             adaptersToBeDeleted.forEach(name -> {
                 try {
-                    log.debug("Deleting adapter '{}'", name);
-                    stopAsync(name, true)
-                            .whenComplete((ignored, t) -> deleteAdapterInternal(name))
-                            .get();
-                } catch (final InterruptedException | ExecutionException e) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Deleting adapter '{}'", name);
+                    }
+                    stopAsync(name, true).whenComplete((ignored, t) -> deleteAdapterInternal(name)).get();
+                } catch (final InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                    failedAdapters.add(name);
+                    log.error("Interrupted while deleting adapter {}", name, e);
+                } catch (final ExecutionException e) {
                     failedAdapters.add(name);
                     log.error("Failed deleting adapter {}", name, e);
                 }
             });
 
             adaptersToBeCreated.forEach(name -> {
                 try {
-                    log.debug("Creating adapter '{}'", name);
-                    startAsync(createAdapterInternal(protocolAdapterConfigs.get(name), versionProvider.getVersion())).get();
-                } catch (final InterruptedException | ExecutionException e) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Creating adapter '{}'", name);
+                    }
+                    startAsync(createAdapterInternal(protocolAdapterConfigs.get(name),
+                            versionProvider.getVersion())).get();
```

Comment:
This makes the adapter creations in sequence. Do we want to run this in parallel? If YES, we could call `map()` to get a list of `Future`s, then call `get()` per future. It'll be much faster.